modules/cluster-compare-templating.adoc

```diff
@@ -0,0 +1,42 @@
+:_mod-docs-content-type: PROCEDURE
+
+[id="cluster-compare-templating_{context}"]
+= Configuring expected variation in a template
+
+You can handle variable content within a template by using Golang templating syntax. Using this syntax, you can configure validation logic that handles optional, required, and conditional content within the template.  
+
+[NOTE]
+====
+You can use the Golang templating syntax with custom and built-in functions for complex use cases. All Golang built-in functions are supported including the functions in the Sprig library. This implementation uses the Helm templating behavior and supports all custom functions that are used in Helm, such as `example: toYaml`.
+====
+
+.Procedure
+
+* Create a `metadata.yaml` file to match your use case. Use the following structure as an example:
++
+[source,yaml]
+----
+apiVersion: v2
+kind: Service
+metadata:
+  name: frontend
```

Comment:
Maybe needs a <4> reference for this line, as an example of a required field where the value must match exactly as specified.

Reply:
Added a callout accordingly